# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/TSFTextStore.cpp
# Commit: 14e68c94b303
# Full Hash: 14e68c94b3036427ed9cac5adca7615b11d1df81
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 1-1: Make `TSFTextStore::Selection` work with `IMENotification::SelectionChangeDataBase` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137404
# ==============================================================================

diff -r 41535e3d6028 -r 14e68c94b303 widget/windows/TSFTextStore.cpp
--- a/widget/windows/TSFTextStore.cpp	Mon Feb 07 22:27:52 2022 +0000
+++ b/widget/windows/TSFTextStore.cpp	Mon Feb 07 22:33:30 2022 +0000
@@ -2812,7 +2812,7 @@
   // TODO: We should avoid to run this hack on fixed builds.  When we get
   //       exact build number, we should get back here.
   static bool sTSFMayCrashIfGetSelectionReturnsError =
-      IsWindows10BuildOrLater(14393);
+      IsWin10AnniversaryUpdateOrLater();
   return sTSFMayCrashIfGetSelectionReturnsError;
 }
 
@@ -6045,15 +6045,8 @@
   // If selection range isn't actually changed, we don't need to notify TSF
   // of this selection change.
   if (mSelectionForTSF.isNothing()) {
-    mSelectionForTSF.emplace(mPendingSelectionChangeData.mOffset,
-                             mPendingSelectionChangeData.Length(),
-                             mPendingSelectionChangeData.mReversed,
-                             mPendingSelectionChangeData.GetWritingMode());
-  } else if (!mSelectionForTSF->SetSelection(
-                 mPendingSelectionChangeData.mOffset,
-                 mPendingSelectionChangeData.Length(),
-                 mPendingSelectionChangeData.mReversed,
-                 mPendingSelectionChangeData.GetWritingMode())) {
+    mSelectionForTSF.emplace(mPendingSelectionChangeData);
+  } else if (!mSelectionForTSF->SetSelection(mPendingSelectionChangeData)) {
     mPendingSelectionChangeData.Clear();
     MOZ_LOG(gIMELog, LogLevel::Debug,
             ("0x%p   TSFTextStore::NotifyTSFOfSelectionChange(), "